### DISCUSSION TITLE
Send email or other method

### DISCUSSION THREAD

**Discussion by Smiley-k** [authelia/authelia#6105] [Ideas]
Hi, you'll probably say it's not safe, but I'll still try to defend the email method.  )

When using two-factor sending of notifications to users, it is possible to use both corporate mail, which would be the most secure organization if there is no permission to send outside emails, and to use the method of delivery to the user's personal mail, which would ensure the safety of the user in the company.

Teaching users to configure 2fa for themselves on a cell phone can be a big quest when 50-100-200 users, and why if there is already a corporate email ?! which itself sends the user to his personal email otp code and no problem. (or to the user's personal e-mail, which is specified in the user's profile for delivery of such messages)

p/s
It would also be very convenient to perform delivery by any other method, for example, by local services of our own as https://github.com/caronc/apprise   When a company chooses for itself a convenient method of delivery of otp codes, it will greatly reduce the burden of integration with various services - look at this really very cool project.

or https://github.com/binwiederhier/ntfy only push method

**Comment by james-d-elliott** [comment]
As previously stated I think it's something we had in mind, and have made efforts recently which would assist in this in the future.

**Reply by Smiley-k** [reply]
Excuse me, but can I get acquainted with the plans? I am very interested - if it is possible of course.

**Comment by skug67** [comment]
Totally agreed this would be a great feature -- and that either ntfy or apprise would be great technology to leverage
